x-pack/plugins/features/common/feature.ts

```diff
@@ -114,7 +124,60 @@ export interface Feature<
   };
 }
 
-export type FeatureWithAllOrReadPrivileges = Feature<{
-  all?: FeatureKibanaPrivileges;
-  read?: FeatureKibanaPrivileges;
-}>;
+export class Feature {
```

Comment:
**note** This class doesn't provide a ton of benefit on its own to the server-side implementation, but it allows the client-side to extend this with security-specific constructs, which allows the UI to reason about secured features in a more logical way, without having to make the Features plugin aware of them